app/src/main/java/com/example/atheneum/activities/ShowRequestInfoActivity.java

```diff
@@ -125,5 +140,108 @@ public void onClick(View v) {
+                            public void onChanged(@Nullable Transaction transaction) {
+                                if(transaction != null) {
+                                    Log.i(TAG, "updateTransaction(): got transaction" + transaction.toString());
+                                    transaction.setBScan(true);
+                                    transaction.setOwnerID(ownerID);
+                                    FirebaseUser currentUser  = FirebaseAuth.getInstance().getCurrentUser();
+                                    Log.i(TAG, "UserID is: " +   currentUser.getUid());
+                                    transaction.setBorrowerID(currentUser.getUid());
+                                    Log.i(TAG, "value of BScan is" + String.valueOf(transaction.getBScan()));
+                                    Log.i(TAG, "value of OScan is" + String.valueOf(transaction.getOScan()));
+                                    transactionViewModel.updateTransaction(transaction);
+
+//                                    scanBook.setVisibility(View.INVISIBLE);
+
+                                    if(transaction.getBScan() && transaction.getOScan()){
+                                        if(transaction.getType().equals("CHECKOUT")) {
+                                            bOok.setStatus(Book.Status.BORROWED);
+                                            DatabaseWriteHelper.updateBook(bOok);
+                                            transaction.setBScan(false);
+                                            transaction.setOScan(false);
+                                            transaction.setType(Transaction.RETURN);
+                                            DatabaseWriteHelper.updateTransaction(transaction);
+                                            scanBook.setVisibility(View.INVISIBLE);
+                                        }
+                                        else {
+                                            bOok.setStatus(Book.Status.AVAILABLE);
+                                            bOok.setBorrowerID("");
+                                            DatabaseWriteHelper.updateBook(bOok);
+                                            DatabaseWriteHelper.deleteTransaction(transaction);
+                                        }
+                                    }
+                                }
+                                transactionLiveData.removeObserver(this);
+                            }
+                        });
+                    }
+
+                }
+                else{
+                    Toast.makeText(this, "Error: Barcode not found",
+                            Toast.LENGTH_SHORT).show();
+                }
+            }
+        }
+        else{
+            Log.i(TAG, "in else");
+            super.onActivityResult(requestCode, resultCode, data);
+        }
+        Log.i(TAG, "leaving onActivityResult");
+    }
+
+    //TODO
+//    @Override
```

Comment:
Dead code?

Reply:
This is where I'm going to delete the request